docs/use_cases/otel/preprocessing.md

```diff
@@ -0,0 +1,153 @@
+# Ingress
+
+To push data into Feldera, we configure the OpenTelemetry (OTel) Collector to send data to Feldera via HTTP.
+
+#### OTel Collector Configuration:
+
+```yml
+exporters:
+
+  # Send traces to the Feldera pipeline.
+  # Currently only uncompressed JSON is supported.
+  otlphttp/feldera_traces:
+     traces_endpoint: http://feldera:8080/v0/pipelines/otel/ingress/otel_traces?format=json&update_format=raw
+     encoding: json
+     compression: none
+
+  otlphttp/feldera_logs:
+     logs_endpoint: http://feldera:8080/v0/pipelines/otel/ingress/otel_logs?format=json&update_format=raw
+     encoding: json
+     compression: none
+
+  otlphttp/feldera_metrics:
+     metrics_endpoint: http://feldera:8080/v0/pipelines/otel/ingress/otel_metrics?format=json&update_format=raw
+     encoding: json
+     compression: none
+
+service:
+   pipelines:
+     traces:
+       exporters: [spanmetrics, otlphttp/feldera_traces]
+
+     logs:
+       receivers: [otlp]
+       exporters: [otlphttp/feldera_logs]
+
+     metrics:
+       receivers: [hostmetrics, docker_stats, httpcheck/frontend-proxy, otlp, prometheus, redis, spanmetrics]
+       exporters: [otlphttp/feldera_metrics]
+```
+
+Notice that we have the query params: `format=json&update_format=raw`
+- `format=json`: Specifies that the input data is JSON-formatted.
+- `update_format=raw`: Indicates that the request body contains raw data, which is processed as an insertion into the table.
+
+## Preprocessing
+
+Next we perform a series of `UNNEST` operations to extract individual rows of metrics, logs and traces from a nested arrays of them.
```

Comment:
you are now shifting to a different box in the dataflow. Say it.

Reply:
this SQL program is to be concatenated with the type declarations shown before.
so add a comment in both programs saying this.

Reply:
explain the high-level point: the data contains arrays of fields, and you want to convert it into traditional database table rows, for easier processing in SQL.